app/routes/changes-enquiries/deferral/functions.js

```diff
@@ -0,0 +1,124 @@
+  const keyDetails = keyDetailsHelper.formatter(awardDetails);
+  if (Object.keys(errors).length === 0) {
+    dataStore.save(req, 'date-request-received', details, 'deferral');
+    res.redirect(`${root}/deferral/default-date`);
+  } else {
+    res.render('pages/changes-enquiries/deferral/date-request-received', {
+      backLink: stopStatePensionUrl,
+      details,
+      errors,
+      formAction: dateRequestReceivedUrl,
+      keyDetails,
+    });
+  }
+}
+
+function getDefaultDate(req, res) {
+  const awardDetails = dataStore.get(req, 'awardDetails');
+  const defaultDate = dataStore.get(req, 'default-date', 'deferral');
+  const keyDetails = keyDetailsHelper.formatter(awardDetails);
+  const { statePensionDate } = dataStore.get(req, 'awardDetails');
+  res.render('pages/changes-enquiries/deferral/default-date', {
+    backLink: dateRequestReceivedUrl,
+    defaultDate,
+    formAction: defaultDateUrl,
+    keyDetails,
+    statePensionDate: dateHelper.longDate(statePensionDate),
+  });
+}
+
+function postDefaultDate(req, res) {
+  const awardDetails = dataStore.get(req, 'awardDetails');
+  const details = req.body;
+  const errors = formValidator.defaultDate(details);
+  const keyDetails = keyDetailsHelper.formatter(awardDetails);
+  const { statePensionDate } = dataStore.get(req, 'awardDetails');
+  if (Object.keys(errors).length === 0) {
+    dataStore.save(req, 'default-date', details['default-date'], 'deferral');
+    if (details['default-date'] === 'yes') {
+      dataStore.save(req, 'from-date', statePensionDate, 'deferral');
+      res.redirect(`${root}/deferral/confirm`);
+    } else {
+      res.redirect(`${root}/deferral/from-date`);
+    }
+  } else {
+    res.render('pages/changes-enquiries/deferral/default-date', {
+      backLink: dateRequestReceivedUrl,
+      errors,
+      formAction: defaultDateUrl,
+      keyDetails,
+      statePensionDate: dateHelper.longDate(statePensionDate),
+    });
+  }
+}
+
+function getConfirm(req, res) {
+  const awardDetails = dataStore.get(req, 'awardDetails');
+  const keyDetails = keyDetailsHelper.formatter(awardDetails);
+  res.render('pages/changes-enquiries/deferral/confirm', {
+    backLink: defaultDateUrl,
+    button: `${root}/deferral/update`,
+    keyDetails,
+  });
+}
+
+function getUpdate(req, res) {
+  const { nino } = dataStore.get(req, 'awardDetails');
+  const fromDate = dataStore.get(req, 'from-date', 'deferral');
+  const deferralDetails = deferralObject.formatter(nino, fromDate);
+  const recordDeferralApiUri = 'api/award/record-deferral';
+  const putCall = requestHelper.generatePutCall(res.locals.agentGateway + recordDeferralApiUri, deferralDetails, 'award', req.user);
+  request(putCall)
+    .then(() => {
+      req.flash('success', i18n.t('deferral-update:messages.success'));
+      deleteSession.deleteAllDeferralSession(req);
+      res.redirect(root);
+    })
+    .catch((err) => {
+      const traceID = requestHelper.getTraceID(err);
```

Comment:
Have a little helper that does all this for you:

```
errorHelper.flashErrorAndRedirect(req, res, err, 'award', `${root}/deferral/update`);
```

Reply:
done